app/square_size.go

```diff
@@ -0,0 +1,29 @@
+package app
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GovMaxSquareSize returns the maximum square size that can be used for a block
+// using the max bytes value from the consensus params. Governance can change
+// the result of this value by changing the MaxBytes consensus parameter.
+func (app *App) GovMaxSquareSize(ctx sdk.Context) int {
+	// TODO: fix hack that forces the max square size for the first height to
+	// 64. This is due to tendermint not technically supposed to be calling
+	// PrepareProposal when heights are not >= 1. This is remedied in versions
+	// of the sdk and coment that have full support of PreparePropsoal, although
+	// celestia-app does not currently use those. see this PR for more
+	// details https://github.com/cosmos/cosmos-sdk/pull/14505
+	if ctx.BlockHeader().Height == 0 {
+		return int(appconsts.DefaultGovMaxSquareSize)
+	}
```

Comment:
pls review, not sure of the best solution around this, but this seemed the most reasonable. 

I think we should also add a check in PrepareProposal that will simply propose an empty block for the first block (height ==0), this way if there are enough transactions in the first block to reach a govMaxSquareSize, we will likely avoid having to encode this logic in future upgrades forever!!